cmd/endorser/README.md

```diff
@@ -1,40 +1,39 @@
-that can be customized via a dedicated input argument.
+Inputs:
+*  `--provenance_uris`: Zero or more provenances, as a comma-separated list of URIs. The tool retrieves the URIs and evaluates them
+*  `--verification_options` Custom verification to run on the provenances, as a prerequisite to the endorsement generation. Optional - if not specified then no verifications are carried out. See the underlying [protocol buffer definition](../../proto/verification_options.proto)
+*  `--binary_name`: The name of the binary
+*  `--binary_path`: Path to the binary file. Needed only to compute digests
 
-If no provenance URIs are provided, the tool generates a provenance-less endorsement statement if
-the given verification options allows that. For more information about verification options, see the
-[protobuf specification](../../proto/provenance_verification.proto).
+Outputs:
+*  `--output_path`: Where the endorsement (a JSON file) goes. Common example: `--output_path=endorsement.json`
 
-If a non-empty list of provenance URIs is provided, the tool downloads them, verifies them according
-to the options in the provided verification options file, and if the verification is successful
-generates an endorsement statement, with the given provenances listed in the endorsement statement
-as evidence (in its evidence field).
-
-Example execution without provenances:
+Here is a simple example which neither involves provenances nor verification:
 
 ```bash
 go run cmd/endorser/main.go \
- --binary_path=testdata/binary \
- --binary_name=stage0_bin \
- --verification_options=testdata/skip_verification.textproto
+  --binary_path=testdata/binary \
+  --binary_name=stage0_bin \
+  --output_path=/tmp/endorsement.json
 ```
 
-Example execution with a provenance URI from ent (for simplicity we pass in
-`testdata/skip_verification.textproto` for verification):
+A more involved example with a single provenance and some verification:
 
 ```bash
 go run cmd/endorser/main.go \
- --binary_path=testdata/binary \
- --binary_name=stage0_bin \
- --provenance_uris=https://ent-server-62sa4xcfia-ew.a.run.app/raw/sha2-256:94f2b47418b42dde64f678a9d348dde887bfe4deafc8b43f611240fee6cc750a \
- --verification_options=testdata/skip_verification.textproto
+  --binary_path=testdata/binary \
+  --binary_name=stage0_bin \
+  --provenance_uris=https://ent-server-62sa4xcfia-ew.a.run.app/raw/sha2-256:94f2b47418b42dde64f678a9d348dde887bfe4deafc8b43f611240fee6cc750a \
+  --verification_options="provenance_count_at_least { count: 1 } all_with_build_command {} all_with_binary_digests { formats: 'sha2-256' digests: '70a4fae8cd01e8e509f0d29efe9cf810192ad9b606fcf66fb6c4cbfee40fd951'}" \
+  --output_path=/tmp/endorsement.json
 ```
 
-See [this comment](https://github.com/project-oak/oak/pull/4191#issuecomment-1643932356) as the
-source of the binary and provenance info.
+If the verification options should be kept in a file (for length reasons), then use
```

Comment:
TBH I think the common case is to keep the verification in a file, so we can just point at the file name, or require setting an explicit `--skip-verification` flag for extra safety.

Who is going to try and embed a textproto in a flag in practice? Even if it is allowed, it seems very error prone :)

Reply:
What if the binary digest is part of the verification, then the textproto changes on every call. Also, it is likely called by another script, so we cut the disk access and the temporary files. Yes, I think it's about calling this directly or by another script. In the former case, you'd prefer passing a file, in the latter case, passing it directly.

It is a long command line if there are many checks, but I really liked verifying the counts directly because it is expressive (OTOH you see the number of URIs passed in the other argument):

--verification_options="provenance_count_at_least { count: 2 }"

--skip_verification is a good idea.

Reply:
Done.